### PR TITLE
Restored MacOS support

### PR DESCRIPTION
**Pull request by SumolX**
Attached is a Universal Binary tested on a M3.  I will need to test on my intel machine as well.  Feedback appreciated!
 
[OpenBOR-MacOS-Universal.zip](https://github.com/DCurrent/openbor/files/14855827/OpenBOR-MacOS-Universal.zip)

### CODE REVIEW

**Comment by DCurrent** [comment]
@SumolX, I don't have a Mac, so there's really no way I can review this. Since you were the primary coder before me, I'll trust your judgement.

File: engine/openbor.c [hunk 1]
```diff
@@ -42353,15 +42353,17 @@ void kill_all_enemies()
 
 void smart_bomb(entity *e, s_attack *attack)    // New method for smartbombs
 {
-    int i, hostile, hit = 0;
+    int i, hit = 0;
     entity *tmpself = NULL;
     s_defense* defense_object = NULL;
 
-    hostile = e->faction.type_hostile;
+#if 0 // TODO: why not referenced in code
```
Comment (DCurrent) [inline engine/openbor.c:42360]: Can you elaborate on this comment?

**Review by DCurrent — COMMENTED** [review]
@SumolX, I don't have a Mac, so there's really no way I can review this. Since you were the primary coder before me, I'll trust your judgement.

**Comment by SumolX** [comment]
Closing this PR as it has been incorporated with PR  301

**Pull request closed without merging**